src/data/apiClient.js

```diff
@@ -13,4 +16,78 @@ const apiClient = getAuthenticatedAPIClient({
   csrfCookieName: configuration.CSRF_COOKIE_NAME,
 });
 
+const clientServerKeyMap = {
+  bio: 'bio',
+  socialLinks: 'social_links',
+  country: 'country',
+  education: 'level_of_education',
+  fullName: 'name',
+  username: 'username',
+  profileImage: 'profile_image',
+  dateJoined: 'date_joined',
+  languageProficiencies: 'language_proficiencies',
+  accountPrivacy: 'account_privacy',
+};
+
+const serverClientKeyMap = _.invert(clientServerKeyMap);
+
+
+export function getPreferences(username) {
+  const url = `${lmsBaseUrl}/api/user/v1/preferences/${username}`;
+
+  return new Promise((resolve, reject) => {
+    apiClient.get(url)
+      .then(({ data }) => {
+        // Unflatten server response
+        // visibility.social_links: 'value' becomes { visibility: { socialLinks: 'value' }}
```

Comment:
@nasthagiri Is it possible to send objects instead of a flat object with dot syntax keys? From:
```
{
   "visibility.social_links": value
}
```

To:
```
{
   visibility: {
      social_links: value
   }
}
```

Reply:
If we created a separate API specifically for, let's say, visibility settings, then we would return it as objects as proposed.  However, we're currently re-using the existing "user preferences" API since that's where visibility preferences have been historically accessed.  So we have extended that API to store account-level visibility preferences.

Keep in mind that "visibility settings" for other "pluggable features", like Course Certificates, would come from a different Course-Certs-specific API.

Reply:
We can discuss this further.